### ISSUE TITLE
[Feature] Ability to upgrade library with upscaling algorithms

### ISSUE THREAD

**Issue by dsrtusr88** [stashapp/stash#3234]
**Is your feature request related to a problem? Please describe.**
Would like the ability to integrate https://github.com/k4yt3x/video2x so that files can be regenerated as an upscaled output file.

**Describe the solution you'd like**
Would like a mode to take a file (ex. 480p scene), set basic parameters (upscale to 720p/1080p/4k), set preferred algorithm (RealSR/SRMD/RealCUGAN), set quality (2x/4x/6x), and then leverage a CPU or GPU to process the output file.  On output, scene compare mode to play both files at the same time with the ability to toggle sound on either and then decide which one to keep.

**Comment by PhuriousGeorge** [comment]
Personally I don't see the point in adding this functionality in an "organizer"/"player". It sounds like what you really want based on your "solution" is a better front-end for their application.

At most, this might be a good case for a third-party (not Stash devs) plugin if integration is desired.

**Comment by dsrtusr88** [comment]
@PhuriousGeorge i tend to agree, really on the fence on ‘in-scope’, but I think the StashApp has so much more potential as a library/stash optimizer.

**Comment by WithoutPants** [comment]
This was discussed privately around potentially adding a bounty for the desired functionality. I agree that this should be a third-party plugin, but the current plugin system cannot facilitate the use case described, so the plugin system needs to be improved to support something like this. Any bounty would be predicated on improving the plugin system to support accepting parameters and hooking into the scene operations menu.

**Comment by StashPowerUser** [comment]
There are a lot of management type features that would be nice to have in stash in the future that could fit into this model.

Like being able to schedule the transcoding of scenes to x265 on a queue, have the converted file added to the scene as an alternate file, and then allowing you to delete the original if the result is acceptable.

This sort of flow would be great for AI Upscale as well, could potentially be an option for the transcode.

Right now, if I see some scenes I want to transcode (or AI Upscale) I have to copy/symlink the files I want to transform, run some command line app, wait for it to finish, then move the results into my stash and delete the originals, rescan/generate etc... and any tagging I've done on the original files is lost.
